server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/vacation/CassandraVacationDAO.java

```diff
@@ -92,27 +107,24 @@ public CassandraVacationDAO(Session session, CassandraTypesProvider cassandraTyp
                 .getZonedDateTime());
     }
 
-    private Insert createSpecificUpdate(VacationUpdate vacationUpdate, Insert baseInsert) {
-        Insert insertWithSubject = addValue(baseInsert, CassandraVacationTable.SUBJECT, vacationUpdate.getSubject());
-        Insert insertWithHtml = addValue(insertWithSubject, CassandraVacationTable.HTML, vacationUpdate.getHtmlBody());
-        Insert insertWithText = addValue(insertWithHtml, CassandraVacationTable.TEXT, vacationUpdate.getTextBody());
-        Insert insertWithFromDate = addValueZonedDateTime(insertWithText, CassandraVacationTable.FROM_DATE, vacationUpdate.getFromDate());
-        Insert insertWithToDate = addValueZonedDateTime(insertWithFromDate, CassandraVacationTable.TO_DATE, vacationUpdate.getToDate());
-        return addValue(insertWithToDate, CassandraVacationTable.IS_ENABLED, vacationUpdate.getIsEnabled());
+    private Insert createSpecificUpdate(VacationPatch vacationPatch, Insert baseInsert) {
+        return insertGeneratorPipeline
+            .apply(vacationPatch)
+            .apply(baseInsert);
     }
 
-    public <T> Insert addValue(Insert insert, String field, MissableOptional<T> missableValue) {
+    public <T> Function<Insert, Insert> addValue(String field, MissableOptional<T> missableValue) {
         if (!missableValue.isMissing()) {
-            return insert.value(field, missableValue.getValue());
+            return insert -> insert.value(field, missableValue.getValue());
         }
```

Comment:
map is for VALUE state
I can create a map for NULL and VALUE states (that then takes Optional<T> ) and chain with notMissingOrElse

Would you like to see that ?